evm/src/arithmetic/operations.rs

```diff
@@ -0,0 +1,156 @@
+use ethereum_types::U256;
+use plonky2::hash::hash_types::RichField;
+
+use crate::arithmetic::columns::*;
+use crate::arithmetic::{add, compare, modular, mul, sub};
+
+#[inline]
+fn u64_to_array<F: RichField>(out: &mut [F], x: u64) {
+    debug_assert!(out.len() == 4);
+
+    const MASK: u64 = (1 << 16) - 1;
+    out[0] = F::from_canonical_u64(x & MASK);
+    out[1] = F::from_canonical_u64((x >> 16) % MASK);
+    out[2] = F::from_canonical_u64((x >> 32) % MASK);
+    out[3] = F::from_canonical_u64((x >> 48) % MASK);
+}
+
+fn u256_to_array<F: RichField>(out: &mut [F], x: U256) {
+    debug_assert!(out.len() == N_LIMBS);
+
+    u64_to_array(&mut out[0..4], x.0[0]);
+    u64_to_array(&mut out[4..8], x.0[1]);
+    u64_to_array(&mut out[8..12], x.0[2]);
+    u64_to_array(&mut out[12..16], x.0[3]);
+}
+
+pub trait Operation<F: RichField> {
+    /// Convert operation into one or two rows of the trace.
+    ///
+    /// Morally these types should be [F; NUM_ARITH_COLUMNS], but we
+    /// use vectors because that's what utils::transpose expects.
+    fn to_rows(&self) -> (Vec<F>, Option<Vec<F>>);
+}
```

Comment:
Is the goal the remove this trait at some point? If not, maybe it would be worth renaming to avoid the name collision with `arithmetic::Operation`.

Reply:
Yeah, I mentioned in the PR description, but not very clearly, that I think this `Operation` trait should be merged with `arithmetic::Operation` in an upcoming PR (the latter trait appeared unbeknownst to me while I was writing this PR, hence the clash).